src/main/java/seedu/address/model/grades/GradesManager.java

```diff
@@ -0,0 +1,162 @@
+     */
+    public void addGrade (Grades grade) {
+        grades.add(grade);
+    }
+
+    /**
+     This method shows grades of students for one grade component in module.
+     */
+    public void createGraph (LineChart<Number, Number> lineChart, Grades grade) {
+        XYChart.Series<Number, Number> series = new XYChart.Series<>();
+
+        // hashmap to store the frequency of element
+        Map<Float, Integer> hashMapOfGrades = new HashMap<>();
+        for (Grades gradesList : grades) {
+            if (gradesList.getModuleCode().equals(grade.getModuleCode())
+                    && gradesList.getGradeComponentName().equals(grade.getGradeComponentName())) {
+                Integer count = hashMapOfGrades.get(gradesList.getMarks());
+                hashMapOfGrades.put(gradesList.getMarks(), (count == null) ? 1 : count + 1);
+            }
+        }
+        // displaying the occurrence of elements in the arraylist
+        for (Map.Entry<Float, Integer> val : hashMapOfGrades.entrySet()) {
+            series.getData().add(new XYChart.Data<>(val.getKey(), val.getValue()));
+        }
+        lineChart.getData().add(series);
+    }
+
+
+    /**
+     This method checks if module code and component name have empty inputs.
+     */
+    public boolean isEmpty (String moduleCode, String gradebookComponentName, String adminNo) {
+        boolean isEmpty = false;
+        if (moduleCode.equals("") || gradebookComponentName.equals("") || adminNo.equals("")) {
+            isEmpty = true;
+        }
+        return isEmpty;
+    }
+
+    /**
+     This method checks if admin no already assigned a grade.
+     */
+    public boolean isDuplicate (String moduleCode, String gradebookComponentName, String adminNo) {
+        boolean isDuplicate = false;
+        Grades grade = findAdminNo(moduleCode, gradebookComponentName, adminNo);
+        if (grade != null) {
+            isDuplicate = true;
+        }
+        return isDuplicate;
+    }
+
+    /**
+     This method finds admin no to an assigned grade item in Trajectory.
+     */
+    public Grades findAdminNo (String moduleCode, String gradebookComponentName, String adminNo) {
+        for (Grades grade : grades) {
+            if (grade.getModuleCode().equals(moduleCode)
+                    && grade.getGradeComponentName().equals(gradebookComponentName)
+                    && grade.getAdminNo().equals(adminNo)) {
+                return grade;
+            }
+        }
+        return null;
+    }
+
+    /**
+     This method checks if marks are within acceptable range.
+     */
+    public boolean isMarksValid (float studentMarks) {
+        boolean isMarksValid = false;
+        if (studentMarks >= 0 && studentMarks <= 100) {
+            isMarksValid = true;
+        }
+        return isMarksValid;
+    }
+
+    /**
+     This method checks if student is enrolled to module.
+     */
+    public boolean isStudentEnrolledToModule (String moduleCode, String adminNo) {
```

Comment:
Enrolling of students into a module is part of my feature, so this API should be provided by my manager class. Please switch to it when I have pushed it out

Reply:
Will try to make use of this by v1.3, if not will be pushed to v1.4 release. Thanks